nova/virt/libvirt/volume/mount.py

```diff
@@ -385,9 +387,17 @@ def _real_umount(self, mountpoint):
             LOG.error("Couldn't unmount %(mountpoint)s: %(reason)s",
                       {'mountpoint': mountpoint, 'reason': str(ex)})
 
-        if not os.path.ismount(mountpoint):
-            nova.privsep.path.rmdir(mountpoint)
-            return False
+        for retries in range(CONF.libvirt.num_umount_retries, -1, -1):
```

Comment:
```suggestion
        for tries in reversed(range(CONF.libvirt.num_umount_retries + 1)):
```

Would read easier and less "clever" for me. Also these are then technically tries, not just retries.

Reply:
Done. I had to think about it a while, because `range(CONF.libvirt.num_umount_retries, -1, -1)` comes more intuitively to me.